service/history/archival/archiver_test.go

```diff
@@ -181,7 +189,11 @@ func TestArchiver(t *testing.T) {
 			sdkClientFactory := sdk.NewMockClientFactory(controller)
 			sdkClientFactory.EXPECT().GetSystemClient().Return(sdkClient).AnyTimes()
 
-			historyURI, err := carchiver.NewURI("test:///history/archival")
+			var historyURI carchiver.URI
+			var err error
+			if !c.NilHistoryUri {
```

Comment:
This is a bit crude because if we just did `c.HistoryURI` - we wouldn't know whether to use default or if it's unset